app/src/lib/stores/app-store.ts

```diff
@@ -2379,7 +2379,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
     return this.withIsCommitting(repository, async () => {
       const result = await gitStore.performFailableOperation(async () => {
         const message = await formatCommitMessage(repository, context)
-        return createCommit(repository, message, selectedFiles) !== undefined
+        return createCommit(repository, message, selectedFiles)
       })
 
       if (result) {
```

Comment:
Since result is a string (`string | undefined`) here now we should be explicit or else an empty string and `undefined` will be treated the same.

```suggestion
      if (result  !== undefined) {
```